src/Harald.WebApi/Infrastructure/Messaging/EventDispatcher.cs

```diff
@@ -27,20 +28,37 @@ public class EventDispatcher : IEventDispatcher
 
         public async Task Send(string generalDomainEventJson, IServiceScope serviceScope)
         {
-            var generalDomainEventObj = JsonConvert.DeserializeObject<GeneralDomainEvent>(generalDomainEventJson);
+            GeneralDomainEvent generalDomainEventObj = new GeneralDomainEvent("", "", "", "", "");
+            try
+            {
+                generalDomainEventObj = JsonConvert.DeserializeObject<GeneralDomainEvent>(generalDomainEventJson);
+            }
+            catch (JsonReaderException ex)
+            {
+                throw new EventMessageIncomprehensible($"Received a message that could not be deserialized from expected JSON structure. Original exception: {ex}");
+            }
             await SendAsync(generalDomainEventObj, serviceScope);
```

Comment:
Move this into try-part, and avoid the "New GeneraltDomainEvent(.......) ctor.

Reply:
By "this" I assume you mean line 40? I can't quite see what the purpose of that would be. The try-catch was added in order to ensure the handling of that specific line, and the message given to EventMessageIncomprehensible was made for just that potential error that might occur there. The scope that line 40 is running in, is already wrapped in a try-catch at line 59 in `Infrastructure/Messaging/ConsumerHostedService.cs`. Perhaps I'm missing something?

Regarding "_avoid the "New GeneraltDomainEvent(.......) ctor._" - I assume you mean the one at line 31. That was mostly done in order to avoid having something return null and the caller not being able to handle that.
It's usually different from team to team how they interpret when one should use null. Since I was unable to find any guidelines, I just went with something.

Is the following an acceptable solution, assuming I'm not misinterpreting anything?
```
GeneralDomainEvent generalDomainEventObj;
```

Reply:
Move into try-catch and added some tests, can be seen here - https://github.com/dfds/harald/pull/10/commits/b8c1ec7ced06655a2690678bd668646279d90bbb